mmocr/models/textrecog/backbones/svtr.py

```diff
@@ -0,0 +1,148 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+# Modified from <https://arxiv.org/abs/2205.00159>
+# Adapted from <https://github.com/PaddlePaddle/PaddleOCR>
+from typing import Callable
+
+import torch.nn as nn
+from mmengine.model import BaseModule
+
+
+class Conv_BN_Layer(BaseModule):
+    """The unit layer of conv3x3 and BatchNorm.
+
+    Args:
+        in_channels (int): Number of input channels.
+        out_channels (int): Number of output channels.
+        kernel_size (int, optional): Number of kernel size.
+        stride (int, optional): The stride for the cross-correlation.
+        padding (int, optional): The amount of padding applied to the input
+        bias (bool, optional): Use or not bias parameters. Defaults to True.
+        groups (int, optional): The amount of group convolution.
+        act (Callable, optional): The activation function.
+    """
+
+    def __init__(self,
+                 in_channels: int,
+                 out_channels: int,
+                 kernel_size: int = 3,
+                 stride: int = 2,
+                 padding: int = 1,
+                 bias: bool = False,
+                 groups: int = 1,
+                 act: Callable = nn.GELU):
+        super().__init__()
+        self.conv = nn.Conv2d(
+            in_channels=in_channels,
+            out_channels=out_channels,
+            kernel_size=kernel_size,
+            stride=stride,
+            padding=padding,
+            groups=groups,
+            bias=bias)
+        self.norm = nn.BatchNorm2d(out_channels)
+        self.act = act()
+
+    def forward(self, x):
+        """Forward function.
+
+        Args:
+            x (Tensor): A Tensor of shape :math:`(N, C, H, W)`.
+
+        Returns:
+            Tensor: The raw logit tensor. Shape :math:`(N, C, H, W)`.
+        """
+        x = self.conv(x)
+        x = self.norm(x)
+        outputs = self.act(x)
+        return outputs
+
+
+class OverlapPatchEmbed(BaseModule):
+    """Image to the progressive overlapping Patch Embedding.
+
+    Args:
+        input_size (int | tuple): The size of input, which will be used to
+            calculate the out size.
+        in_channels (int): Number of input channels.
+        embed_dims (int): The dimensions of embedding.
+        num_layers (int, optional): Number of Conv_BN_Layer. Defaults to 2.
+    """
+
+    def __init__(self,
+                 img_size=[32, 100],
+                 in_channels=3,
+                 embed_dims=768,
+                 num_layers=2,
+                 init_cfg=None):
```

Comment:
Missing typings and default values. We can take [this](https://github.com/open-mmlab/mmocr/blob/dev-1.x/mmocr/models/textrecog/encoders/sar_encoder.py#L20-L47) as an example